libs/sdk-core/src/breez_services.rs

```diff
@@ -473,14 +500,52 @@ impl BreezServices {
         pair_hash: String,
     ) -> Result<ReverseSwapInfo> {
         let routing_hop = self.lsp_info().await?;
-        self.btc_send_swapper
+        let created_reverse_swap_info = self
+            .btc_send_swapper
             .create_reverse_swap(
                 amount_sat,
                 onchain_recipient_address.clone(),
                 pair_hash,
                 routing_hop.pubkey,
             )
-            .await
+            .await?;
+        info!("rev_swap_info: {created_reverse_swap_info:?}");
+
+        // Trigger the payment in a separate thread
+        // Payment usually returns on completion (settlement) or on error. In our case, settlement
+        // happens as the last step of the reverse swap.
+        let created_reverse_swap_info_clone = created_reverse_swap_info.clone();
+        let node_api_clone = self.node_api.clone();
+        let pay_thread_handle: JoinHandle<Result<Payment>> = tokio::spawn(async move {
+            info!("Paying HODL invoice");
+            node_api_clone
+                .send_payment(created_reverse_swap_info_clone.hodl_bolt11, None)
+                .await
+        });
+
+        // For the max configured duration of a payment:
+        // - either wait to see if the payment fails, or
+        // - regularly poll the Boltz API to check if the status advanced to LockTxMempool
+        tokio::select! {
+            pay_thread_res = pay_thread_handle => {
+                // If the payment call returns, there was an error
+                // When successfully paying the HODL invoice, the call doesn't return until settlement,
+                // which cannot happen now (at the beginning of the reverse swap)
+
+                // TODO It doesn't fail when trying to pay more sats than max_payable?
+                match pay_thread_res {
+                    Ok(Ok(_)) => Err(anyhow!("Payment of HODL invoice unexpectedly returned")),
+                    Ok(Err(e)) => Err(anyhow!("Failed to pay HODL invoice: {e}")),
+                    Err(e) => Err(anyhow!("Failed to wait for payment of HODL invoice: {e}"))
+                }
+            },
+            paid_invoice_res = self.poll_initial_boltz_status_transition(&created_reverse_swap_info.id) => {
```

Comment:
Perhaps it is easier to see the timeout here as another select branch rather inside the `poll_initial_boltz_status_transition` function?

Reply:
Good point, addressed in 5371ca5